### PR TITLE
Chequebook endpoint

### PR DESCRIPTION
**Pull request by metacertain**
The aim of this PR is to add /chequebook/balance & /chequebook/addess to the list of endpoints in the bee debug api

### CODE REVIEW

File: pkg/debugapi/chequebook.go [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+type chequebookBalanceResponse struct {
+	Balance *big.Int `json:"balance"`
+}
+
+func (s *server) chequebookBalanceHandler(w http.ResponseWriter, r *http.Request) {
+	balance, err := s.Chequebook.Balance(r.Context())
+	if err != nil {
+		jsonhttp.InternalServerError(w, errCantChequebookBalance)
+		s.Logger.Debugf("debug api: chequebook balance: %v", err)
+		s.Logger.Error("debug api: can not get chequebook balance")
```
Comment (paxthemax) [inline pkg/debugapi/chequebook.go:27]: Should be: cannot

File: pkg/debugapi/chequebook_test.go [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+	"github.com/ethersphere/bee/pkg/debugapi"
+	"github.com/ethersphere/bee/pkg/jsonhttp"
+	"github.com/ethersphere/bee/pkg/jsonhttp/jsonhttptest"
+
+	"github.com/ethersphere/bee/pkg/settlement/swap/chequebook/mock"
+)
+
+func TestChequebookBalance(t *testing.T) {
+	chequebookBalanceFunc := func(context.Context) (ret *big.Int, err error) {
+		ret = big.NewInt(9000)
```
Comment (paxthemax) [inline pkg/debugapi/chequebook_test.go:24]: Extract constant?

File: pkg/debugapi/chequebook_test.go [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+		return ret, nil
+	}
+
+	testServer := newTestServer(t, testServerOptions{
+		ChequebookOpts: []mock.Option{mock.WithChequebookBalanceFunc(chequebookBalanceFunc)},
+	})
+
+	expected := &debugapi.ChequebookBalanceResponse{
+		Balance: big.NewInt(9000),
```
Comment (paxthemax) [inline pkg/debugapi/chequebook_test.go:33]: Use extracted constant?

File: pkg/debugapi/chequebook.go [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+
+import (
+	"math/big"
+	"net/http"
+
+	"github.com/ethersphere/bee/pkg/jsonhttp"
+)
+
+var (
+	errCantChequebookBalance = "Cannot get chequebook balance"
```
Comment (paxthemax) [inline pkg/debugapi/chequebook.go:15]: Consider: `errChequebookBalance`. It's clear enough IMO.

File: pkg/debugapi/export_test.go [hunk 5]
```diff
@@ -5,23 +5,25 @@
...
+	PeerConnectResponse       = peerConnectResponse
+	PeersResponse             = peersResponse
+	AddressesResponse         = addressesResponse
+	WelcomeMessageRequest     = welcomeMessageRequest
+	WelcomeMessageResponse    = welcomeMessageResponse
+	BalancesResponse          = balancesResponse
+	BalanceResponse           = balanceResponse
+	SettlementResponse        = settlementResponse
+	SettlementsResponse       = settlementsResponse
+	ChequebookBalanceResponse = chequebookBalanceResponse
```
Comment (paxthemax) [inline pkg/debugapi/export_test.go:19]: See above.

File: pkg/settlement/swap/chequebook/mock/chequebook.go [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethersphere/bee/pkg/settlement/swap/chequebook"
+)
+
+// Service is the mock settlement service.
+type Service struct {
+	chequebookBalanceFunc func(context.Context) (*big.Int, error)
+}
+
+// WithsettlementFunc sets the mock settlement function
```
Comment (paxthemax) [inline pkg/settlement/swap/chequebook/mock/chequebook.go:21]: Fix comment.

File: pkg/settlement/swap/chequebook/mock/chequebook.go [hunk 7]
```diff
@@ -0,0 +1,61 @@
...
+func WithChequebookBalanceFunc(f func(ctx context.Context) (*big.Int, error)) Option {
+	return optionFunc(func(s *Service) {
+		s.chequebookBalanceFunc = f
+	})
+}
+
+// Newsettlement creates the mock settlement implementation
```
Comment (paxthemax) [inline pkg/settlement/swap/chequebook/mock/chequebook.go:28]: Fix comment.

File: pkg/settlement/swap/chequebook/mock/chequebook.go [hunk 8]
```diff
@@ -0,0 +1,61 @@
...
+func NewChequebook(opts ...Option) chequebook.Service {
+	mock := new(Service)
+	for _, o := range opts {
+		o.apply(mock)
+	}
+	return mock
+}
+
+func (s *Service) Balance(ctx context.Context) (bal *big.Int, err error) {
```
Comment (paxthemax) [inline pkg/settlement/swap/chequebook/mock/chequebook.go:37]: Add docstring comment.

File: pkg/settlement/swap/chequebook/mock/chequebook.go [hunk 9]
```diff
@@ -0,0 +1,61 @@
...
+	if s.chequebookBalanceFunc != nil {
+		return s.chequebookBalanceFunc(ctx)
+	}
+	return big.NewInt(0), errors.New("Error")
+}
+
+func (s *Service) Deposit(ctx context.Context, amount *big.Int) (hash common.Hash, err error) {
```
Comment (paxthemax) [inline pkg/settlement/swap/chequebook/mock/chequebook.go:44]: Add docstring comment.

File: pkg/settlement/swap/chequebook/mock/chequebook.go [hunk 10]
```diff
@@ -0,0 +1,61 @@
...
+	return common.Hash{}, errors.New("Error")
+}
+func (s *Service) WaitForDeposit(ctx context.Context, txHash common.Hash) error {
```
Comment (paxthemax) [inline pkg/settlement/swap/chequebook/mock/chequebook.go:47]: Add docstring comment.

File: pkg/settlement/swap/chequebook/mock/chequebook.go [hunk 11]
```diff
@@ -0,0 +1,61 @@
...
+	return errors.New("Error")
+}
+func (s *Service) Address() common.Address {
```
Comment (paxthemax) [inline pkg/settlement/swap/chequebook/mock/chequebook.go:50]: Add docstring comment.

File: pkg/debugapi/chequebook.go (hunk 4 above)
Comment (paxthemax) [reply pkg/debugapi/chequebook.go:15]: Error string should not be capitalized: https://github.com/golang/go/wiki/CodeReviewComments#error-strings

**Review by paxthemax — APPROVED** [review]
Some very minor style remarks to consider. Approved.

**Review by janos — APPROVED** [review]

**Pull request merged**